Running an action deployed in a private repository doesn't work in the manner described in the docs

Deploying the example action in a private repository (with adjusted visibility, according to the instructions) and calling it from another private repo under the same user, results in an error:

```Error:  [2:875] [ERROR] Could not resolve dependencies: The following artifacts could not be resolved: <groupId>:<artifactId>:jar:<version>(absent): Could not find <groupId>:<artifactId>:jar:<version> in quarkus-github-action (https://maven.pkg.github.com/<user/org>/<repo>)```

The problem is, that calling the action with an automatically generated token `github-token: ${{ secrets.GITHUB_TOKEN }}` doesn't seem to work in a private repository.

Instead you need to create/use a personalised access token with `read:packages` scope, set it as secret in the settings of the repo calling the action, and reference it from your workflow e.g. `github-token: ${{ secrets.MY_ACTION_TOKEN }}`.

When following this procedure, the action succeeds.

If desired, I could create a PR, with proposed changes to the documentation.

--- a/docs/modules/ROOT/pages/create-github-action.adoc
+++ b/docs/modules/ROOT/pages/create-github-action.adoc
@@ -338,18 +338,20 @@ The token is required when consuming the GitHub REST or GraphQL APIs so, in most
 [[private_repo]]
 [WARNING]
 ====
-The workflow above will malfunction when used in a private repository. The automatically generated `GITHUB_TOKEN` only grants access to packages, that are part of the https://docs.github.com/en/packages/learn-github-packages/about-permissions-for-github-packages#access-tokens[same repository] the workflow is defined in.
+You may call a Quarkus GitHub Action defined in a public repository from a public or private GitHub repository.
 
-To use a _GitHub Action_ defined in a separate private repository, belonging to the same user or organization, please follow these steps:
+However, the workflow above will malfunction when the Quarkus GitHub Action is hosted in a private repository as the GitHub hosted Maven repository might not be accessible.
+The automatically generated `GITHUB_TOKEN` only grants access to private packages that are part of the https://docs.github.com/en/packages/learn-github-packages/about-permissions-for-github-packages#access-tokens[same repository] the workflow is defined in.
 
-- Create or use a https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens[Personal Access Token] with `read:packages` scope. Depending on the actions to be performed, additional permissions can be necessary.
+To use a Quarkus GitHub Action hosted in a separate private repository, please follow these steps:
 
+- Create or use a https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens[Personal Access Token] with the `read:packages` scope. Depending on the GitHub API calls to be performed, additional permissions can be necessary.
 - In the repository, that is using the action, https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions[create a secret] using the token.
+- The name you give to the secret (e.g. `MY_ACTION_TOKEN`) must in turn be referenced in the workflow configuration above: Replace &#x2780; with `github-token: ${{ secrets.MY_ACTION_TOKEN }}`.
 
-- The name you give the secret (e.g. `MY_ACTION_TOKEN`) must in turn be referenced in the workflow configuration above: Replace  &#x2780; with `github-token: ${{ secrets.MY_ACTION_TOKEN }}`.
+After applying these steps, the workflow should be executed flawlessly.
 
-After applying these steps, the workflow should be executed without any problems.
-
+It is important to mention though that secrets are not made available in workflows triggered by the `pull_request` event.
 ====
 
 == Next steps

--- a/docs/modules/ROOT/pages/create-github-action.adoc
+++ b/docs/modules/ROOT/pages/create-github-action.adoc
@@ -335,6 +335,23 @@ Create a workflow in a separate repository with the following step:
 <1> `${{ secrets.GITHUB_TOKEN }}` is a temporary secret created by GitHub when initializing the workflow, you don't need to define it yourself.
 The token is required when consuming the GitHub REST or GraphQL APIs so, in most cases, you need to pass it to the action.
 
+[[private_repo]]
+[WARNING]
+====
+The workflow above will malfunction when used in a private repository. The automatically generated `GITHUB_TOKEN` only grants access to packages, that are part of the https://docs.github.com/en/packages/learn-github-packages/about-permissions-for-github-packages#access-tokens[same repository] the workflow is defined in.
+
+To use a _GitHub Action_ defined in a separate private repository, belonging to the same user or organization, please follow these steps:
+
+- Create or use a https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens[Personal Access Token] with `read:packages` scope. Depending on the actions to be performed, additional permissions can be necessary.
+
+- In the repository, that is using the action, https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions[create a secret] using the token.
+
+- The name you give the secret (e.g. `MY_ACTION_TOKEN`) must in turn be referenced in the workflow configuration above: Replace  &#x2780; with `github-token: ${{ secrets.MY_ACTION_TOKEN }}`.
+
+After applying these steps, the workflow should be executed without any problems.
+
+====
+
 == Next steps
 
 Obviously, you can do a lot more with Quarkus GitHub Action and we describe all its features in the xref:developer-reference.adoc[Developer reference].

--- a/docs/modules/ROOT/pages/developer-reference.adoc
+++ b/docs/modules/ROOT/pages/developer-reference.adoc
@@ -70,6 +70,13 @@ With this kind of setup, you can call your action with:
           github-token: ${{ secrets.GITHUB_TOKEN }}
 ----
 
+[TIP]
+====
+To call an action from a private repository, the action itself needs to be in a private repository as well (under the same organization or user).
+
+For details on how to call the action, please see xref:create-github-action.adoc#private_repo[Run your action].
+====
+
 === Named actions
 
 It might be practical for you to develop one GitHub Action for several related purposes.

